algorithms/symmetry/cosym/target.py

```diff
@@ -370,31 +370,30 @@ def compute_gradients(self, x: np.ndarray) -> np.ndarray:
           f: The value of the target function at coordinates `x`.
           grad: The gradients of the target function with respect to the parameters.
         """
-        grad = np.empty(x.shape)
         if self.wij_matrix is not None:
             wrij_matrix = np.multiply(self.wij_matrix, self.rij_matrix)
         else:
             wrij_matrix = self.rij_matrix
 
-        coords = []
         NN = x.size // self.dim
-        for i in range(self.dim):
-            coords.append(x[i * NN : (i + 1) * NN])
+        coords = x.reshape((self.dim, NN))
 
         # term 1
+        grad = np.empty(coords.shape)
         for i in range(self.dim):
-            grad[i * NN : (i + 1) * NN] = np.matmul(wrij_matrix, coords[i])
+            grad[i] = np.matmul(wrij_matrix, coords[i])
```

Comment:
`matmul` can also take stacks of matrices and would interpret it as such if `coords` dimensions is > 2. I think 
`grad = np.matmul(wrij_matrix, coords)` _should_ also work

Maybe for future considerations